### DISCUSSION TITLE
Packaging the Source Code to Executable

### DISCUSSION THREAD

**Discussion by Nazchanel** [ThioJoe/YT-Spammer-Purge#989] [Q&A]
I am interested in packaging a Python-based project into a single executable in the form of an `exe` file.

I am aware of PyInstaller, but am encountering errors with it.

Since this project is based in Python and has been put in the form of an executable I want to ask how this is done.

How was it done in this case @ThioJoe ?

**Comment by ThioJoe** [comment]
Pyinstaller is indeed what I use, but it can be really finicky to get set up with some packages. What type of errors are you seeing?

**Comment by darsh5921** [comment]
Hello , I am too trying to convert the source code into Exe file but getting Z3 error from pyinstaller and getting pywintypes311.dll dependency error by converting latest 2.17 beta-6 version , I also tried to make exe with old 2.16 stable version but getting the same error 

![Screenshot_20230219_095851](https://user-images.githubusercontent.com/104684690/219946269-baea87f7-e78c-467d-a38d-ab2822aa3b2d.png)
![Screenshot_20230212_062909](https://user-images.githubusercontent.com/104684690/219946328-33d1f21e-c544-4e6b-b260-189221c765ee.png)

I Have also tried to man
![Screenshot_20230218_113721](https://user-images.githubusercontent.com/104684690/219946531-e740799f-a547-44fd-90ec-635bb65ac665.png)
ually Put the Spamdomainslist.txt files in Spam Purge resources folder manually but still the app gets more further errors like f3 error for default config file 


Can you please provide the exact command of pyinstaller you used to convert the source code to exe 
also When i converted the exe my exe was around 25 MB while yours provided exe was just 13 mb so can you please tell me the pyinstaller  command
I need it to convert the program and use it in my project , so please reply early if possible
